app/components/forms/login-form.js

```diff
@@ -38,70 +39,73 @@ export default Component.extend(FormMixin, {
         }
       }
     };
-  },
+  }
 
-  actions: {
-    submit() {
-      this.onValid(() => {
-        let credentials = this.getProperties('identification', 'password'),
-            authenticator = 'authenticator:jwt';
+  @action
+  async submit() {
+    this.onValid(async() => {
+      let credentials = this.getProperties('identification', 'password'),
+          authenticator = 'authenticator:jwt';
+      this.setProperties({
+        errorMessage : null,
+        isLoading    : true
+      });
+      try {
+        await this.session.authenticate(authenticator, credentials);
+        const tokenPayload = this.authManager.getTokenPayload();
+        if (tokenPayload) {
+          this.authManager.persistCurrentUser(
+            await this.store.findRecord('user', tokenPayload.identity)
+          );
 
-        this.set('errorMessage', null);
-        this.set('isLoading', true);
+        }
+      } catch (e) {
+        if (e.error) {
+          this.set('errorMessage', this.l10n.tVar(e.error));
+        } else {
+          this.set('errorMessage', this.l10n.t('An unexpected error occurred.'));
+        }
+      }
 
-        this.session
-          .authenticate(authenticator, credentials)
-          .then(async() => {
-            const tokenPayload = this.authManager.getTokenPayload();
-            if (tokenPayload) {
-              this.authManager.persistCurrentUser(
-                await this.store.findRecord('user', tokenPayload.identity)
-              );
-            }
-          })
-          .catch(reason => {
-            if (!(this.isDestroyed || this.isDestroying)) {
-              if (reason && reason.hasOwnProperty('status_code') && reason.status_code === 401) {
-                this.set('errorMessage', this.l10n.t('Your credentials were incorrect.'));
-              } else {
-                this.set('errorMessage', this.l10n.t('An unexpected error occurred.'));
-              }
-              this.set('isLoading', false);
-            } else {
-              console.warn(reason);
-            }
-          })
-          .finally(() => {
-            if (!(this.isDestroyed || this.isDestroying)) {
-              this.set('password', '');
-            }
+      if (!(this.isDestroyed || this.isDestroying)) {
+        this.setProperties(
+          {
+            password  : '',
+            isLoading : false
           });
-      });
-    },
+      }
+    });
+  }
 
-    async auth(provider) {
+  @action
+  async auth(provider) {
+    if (provider === 'facebook') {
       try {
-        if (provider === 'facebook') {
-          this.loader.load('/auth/oauth/facebook')
-            .then(async response => {
-              window.location.replace(response.url);
-            });
+        let response = await this.loader.load('/auth/oauth/facebook');
+        window.location.replace(response.url);
```

Comment:
@CosmicCoder96 Do we need it for now, as O-auth is not implemented properly on server